ftml/src/parsing/rule/impls/block/parser.rs

```diff
@@ -334,18 +339,54 @@ where
                 self.get_optional_space()?;
 
                 // Try to get the argument key
-                // Determines if we stop or keep parsing
-                let current = self.current();
-                let key = match current.token {
-                    Token::Identifier => current.slice,
-                    Token::RightBlock => break,
-                    _ => {
-                        return Err(
-                            self.make_warn(ParseWarningKind::BlockMalformedArguments)
-                        )
+                // Allows any token that matches the regular expression
+                // i.e., alphanumeric, dash, or underscore
+                //
+                // This logic determines if we stop or keep getting arguments
+                //
+                // We could use collect_text_keep() here, but it messes with
+                // get_head_block() so we just have it inline. Also it's a bit
+                // strange since one of the outcomes is to break out of the loop.
+
+                let key = {
+                    let start = self.current();
+                    let mut args_finished = false;
+
+                    loop {
+                        let current = self.current();
+                        match current.token {
+                            // End parsing block head
+                            Token::RightBlock => {
+                                args_finished = true;
+                                break;
+                            }
+
+                            // End parsing argument key
+                            Token::Whitespace | Token::Equals => break,
```

Comment:
Would the distinction between this and end-of-block allow for boolean attributes like a `[[video autoplay]]` that maps to `<video autoplay>`? Not that it matters, because `<video autoplay="true">` is equivalent

Reply:
No, this exists because we consume an optional whitespace afterwards, and the key ends either when it hits ` ` or `=`.

Presently all arguments require an associated value.